interview-questions.md

```diff
@@ -6,52 +6,416 @@ First, without external resources. Challenge yourself to answer from memory as i
+
+In summary, the .map() method is a powerful function in JavaScript that allows developers to easily transform each element in an array. It requires a callback function as its first parameter, which is called for each element in the array and must return a new value. The second parameter, this, is optional and is used to set the context for the callback function.
+
+Source:
+
+"JavaScript Map – How to Use the JS .map() Function (Array Method)" byy Nathan Sebhastian - https://www.freecodecamp.org/news/javascript-map-how-to-use-the-js-map-function-array-method/
+
+-------
+
+### 3. What is the difference between map and filter?
 
 Your answer:
 
+The main difference between the map() and filter() methods in JavaScript is that map() transforms each element in an array into a new value, while filter() creates a new array with only the elements that pass a certain condition. In other words, map() transforms the entire array, while filter() only selects specific elements from the array based on a condition.
+
 Researched answer:
 
-1. What is iteration?
+The map() and filter() methods are both important tools in a JavaScript developer's toolbox when working with arrays.
+
+map() is used when you want to transform each element of an array into a new value. It creates a new array with the same length as the original, where each element corresponds to the result of the transformation function applied to the element of the original array. The original array is not modified by the map() method. It is a pure function that does not mutate the original array. The map() method requires a callback function as an argument, which takes the current element of the array as an argument and returns the new value for that element.
+
+On the other hand, filter() is used when you want to create a new array from the original array, containing only the elements that pass a certain condition. It creates a new array with only the elements that pass the test function. The original array is not modified by the filter() method. It is also a pure function that does not mutate the original array. The filter() method requires a callback function as an argument, which takes the current element of the array as an argument and returns a boolean value indicating whether the element should be included in the new array.
+
+Both map() and filter() methods can be chained together with other array methods, allowing for more complex operations to be performed on arrays in a concise and readable manner. While the main difference between map() and filter() is that the former transforms each element into a new value and the latter selects only certain elements, both methods are powerful and versatile and can be used to solve a wide range of problems when working with arrays.
+
+Source: 
+
+SitePoint: "JavaScript Map vs. Filter: Simplify your JavaScript Code" - https://www.sitepoint.com/map-vs-filter-javascript/
```

Comment:
I suggest you start a blog with your answers. It is quite easy to share on linkedIn. I also recommend sharing with medium.com. These answers could be very helpful to new developers as well a way for you to get exposure in the web development community.